src/AspNet.Security.OAuth.LinkedIn/LinkedInAuthenticationHandler.cs

```diff
@@ -58,6 +60,32 @@ public class LinkedInAuthenticationHandler : OAuthHandler<LinkedInAuthentication
             }
 
             var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+
+            if (Options.Fields.Contains(LinkedInAuthenticationConstants.EmailAddressField))
+            {
+                var emailAddressRequest = new HttpRequestMessage(HttpMethod.Get, Options.EmailAddressEndpoint);
```

Comment:
Can you factor out this additional HTTP request into a private method please? I believe the GitHub provider does a similar thing as it also does an extra call to fetch the email address(es).

Reply:
I did it like the GitHub provider, with a protected virtual method